client/src/components/Sharing/Sharing.vue

```diff
@@ -0,0 +1,469 @@
+            this.errors = Array.from(new Set(this.errors).add(newError));
+        },
+        onCopy() {
+            copy(this.itemUrl);
+            this.tooltipClipboard = "Copied!";
+        },
+        onCopyOut() {
+            this.tooltipClipboard = "Copy URL";
+        },
+        onEdit() {
+            this.showUrl = false;
+        },
+        assignItem(newItem) {
+            if (newItem.errors) this.errors = newItem.errors;
+            this.item = newItem;
+            if (!this.item.extra || newItem.errors.length > 0) {
+                this.item.extra = defaultExtra();
+            }
+
+            this.ready = true;
+        },
+        onChange(newSlug) {
+            this.showUrl = true;
+            const requestUrl = `${this.slugUrl}`;
+            axios
+                .put(requestUrl, {
+                    new_slug: newSlug,
+                })
+                .then((response) => {
+                    this.errMsg = null;
+                    this.assignItem(response.data);
+                    this.item.username_and_slug = `${this.itemSlugParts[0]}${newSlug}`;
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        onImportable(importable) {
+            if (importable) {
+                this.setSharing(this.actions.enable_link_access);
+            } else {
+                this.item.published = false;
+                this.setSharing(this.actions.disable_link_access);
+            }
+        },
+        onPublish(published) {
+            if (published) {
+                this.item.importable = true;
+                this.setSharing(this.actions.publish);
+            } else {
+                this.setSharing(this.actions.unpublish);
+            }
+        },
+        getModel() {
+            this.ready = false;
+            axios
+                .get(`${getAppRoot()}api/${this.pluralNameLower}/${this.id}/sharing`)
+                .then((response) => this.assignItem(response.data))
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setUsername() {
+            const Galaxy = getGalaxyInstance();
+            axios
+                .put(`${getAppRoot()}api/users/${Galaxy.user.id}/information/inputs`, {
+                    username: this.newUsername || "",
+                })
+                .then((response) => {
+                    this.hasUsername = true;
+                    this.getModel();
+                })
+                .catch((error) => this.addError(error.response.data.err_msg));
+        },
+        setSharing(action, user_id, share_option) {
+            let user_ids = undefined;
+            if (user_id) {
+                if (user_id.includes(",")) {
+                    user_ids = user_id.replace(/ /g, "").split(",");
+                } else user_ids = [user_id];
+            }
```

Comment:
```suggestion
            const user_ids = user_id ? user_id.replace(/ /g, "").split(",") : undefined;
```
Neither space nor comma are allowed in user ids or emails, so we don't need the if/else.

Reply:
yeah, this one line solution is better, thanks :+1: